include/onnxruntime/core/session/onnxruntime_cxx_api.h

```diff
@@ -392,46 +407,82 @@ inline ONNXTensorElementDataType TensorTypeAndShapeInfo::GetElementType() const
 }
 
 inline size_t TensorTypeAndShapeInfo::GetDimensionsCount() const {
-  return OrtGetNumOfDimensions(p_);
+  return OrtGetDimensionsCount(p_);
 }
 
 inline void TensorTypeAndShapeInfo::GetDimensions(int64_t* values, size_t values_count) const {
   OrtGetDimensions(p_, values, values_count);
 }
 
 inline std::vector<int64_t> TensorTypeAndShapeInfo::GetShape() const {
-  std::vector<int64_t> output;
-  output.resize(GetDimensionsCount());
-  GetDimensions(output.data(), output.size());
-  return output;
+  std::vector<int64_t> out;
+  out.resize(GetDimensionsCount());
+  GetDimensions(out.data(), out.size());
+  return out;
 }
 
 inline Unowned<TensorTypeAndShapeInfo> TypeInfo::GetTensorTypeAndShapeInfo() const {
   return Unowned<TensorTypeAndShapeInfo>{const_cast<OrtTensorTypeAndShapeInfo*>(OrtCastTypeInfoToTensorInfo(p_))};
 }
 
-inline Value Value::CreateTensor(const OrtAllocatorInfo* info, void* p_data, size_t p_data_len, const int64_t* shape, size_t shape_len,
+inline Value Value::CreateTensor(const AllocatorInfo& info, void* p_data, size_t p_data_len, const int64_t* shape, size_t shape_len,
                                  ONNXTensorElementDataType type) {
   OrtValue* out;
   ORT_THROW_ON_ERROR(OrtCreateTensorWithDataAsOrtValue(info, p_data, p_data_len, shape, shape_len, type, &out));
-  return Value(out);
+  return Value{out};
 }
 
+inline Value Value::CreateMap(Value& keys, Value& values) {
+  OrtValue* out;
+  OrtValue* inputs[2] = {keys, values};
+  ORT_THROW_ON_ERROR(OrtCreateValue(inputs, 2, ONNX_TYPE_MAP, &out));
+  return Value{out};
+}
+
+inline Value Value::CreateSequence(std::vector<Value>& values) {
+  OrtValue* out;
+  std::vector<OrtValue*> values_ort{values.data(), values.data() + values.size()};
+  ORT_THROW_ON_ERROR(OrtCreateValue(values_ort.data(), values_ort.size(), ONNX_TYPE_SEQUENCE, &out));
+  return Value{out};
+}  // namespace Ort
+
 inline bool Value::IsTensor() const {
   return OrtIsTensor(p_) != 0;
 }
 
+inline size_t Value::GetCount() const {
+  size_t out;
+  ORT_THROW_ON_ERROR(OrtGetValueCount(p_, &out));
```

Comment:
Are we expected to use C++ exception in the C++ APIs? It is hard to tell how the caller handle the exception. In this specific case, we do not know whether get count failed is a fatal failure overall or just a transit issue like only one predict request impacted but other requests may still fine. So I would prefer using return Status as return and let the caller decide what to do.

[Coding Conversion](https://github.com/microsoft/onnxruntime/blob/master/docs/Coding_Conventions_and_Standards.md)
> * Exceptions
>     * Allowed to throw fatal errors that are expected to result in a top level handler catching them, logging them and terminating the program.

Reply:
Yes, all of the C++ APIs don't return the OrtStatus* values directly, they convert them into exceptions. You can always try/catch(const Ort::Exception &) around places you expect to fail due to causes under your control. Otherwise you can always use the C API directly if you want to avoid the C++ features.